src/main/java/org/springframework/data/couchbase/core/convert/MappingCouchbaseConverter.java

```diff
@@ -148,6 +148,8 @@ public MappingCouchbaseConverter(
 		this.mappingContext = mappingContext;
 		typeMapper = new DefaultCouchbaseTypeMapper(typeKey != null ? typeKey : TYPEKEY_DEFAULT);
 		spELContext = new SpELContext(CouchbaseDocumentPropertyAccessor.INSTANCE);
+		// ApplicationContext is needed for environment for resolving SPEL expressions.
+		this.setApplicationContext(((CouchbaseMappingContext) mappingContext).getApplicationContext());
```

Comment:
should we throw an exception if the mappingContext is not instanceof CouchbaseMappingContext? So it will be descriptive

Reply:
Actually, the converter should be ApplicationContextAware so that the bean container injects the context. You could skip that line and stop exposing the context from the mapping context for a clean implementation.

Reply:
Yes.  This shouldn't be done here. It's similar to afterPropertiesSet() call, in that it should normally be done by the Bean container. And if the object is created outside the bean container, then the caller has to take care of the initialization that the bean container would normally do.   So I'll remove this and not expose the applicationContext of the mapping context.